### ISSUE TITLE
[wire-service] Remove experimental `WireContextEvent` event name

### ISSUE THREAD

**Issue by caridy** [salesforce/lwc#1357]
As today, we use an experimental mechanism to provide a dummy-context mechanism. This relies on a magic string value `"`WireContextEvent"` that can be used to dispatch a DOM event from a wire adapter.

This is not longer needed, and should be removed in favor of the formal context mechanism specified here https://github.com/salesforce/lwc-rfcs/pull/6.

**Comment by lpomerleau** [comment]
This change, which updated the `BaseLightningElement` event name/type validation to throw an error rather than a warning, essentially disables the `WireContextEvent` mechanism:

https://github.com/salesforce/lwc/pull/1193/files#r280001098

Can it be changed back into a warning until the `WireContextEvent` code is officially removed? Or is it preferred that consumers stick with an older version of the engine until they switch to the new LWC context service?

**Comment by jflowers45** [comment]
Hi, not sure if related, but if debug mode is turned on, any time you wire the CurrentPageReference you get an error. Here's an example that happens in the lwc-recipes.

Invalid event type "WireContextEvent" dispatched in element <c-pubsub-search-bar>.

**Comment by caridy** [comment]
@jflowers45 no, your specific question was answered here: https://github.com/salesforce/lwc/issues/1591#issuecomment-545988069, and this issue is pending to be closed as soon as we merge https://github.com/salesforce/lwc/pull/1459, which should happen in the next few days, with that change, there will be no more `WireContextEvent`.
